presto-spi/src/main/java/com/facebook/presto/spi/plan/PartitioningScheme.java

```diff
@@ -37,16 +37,30 @@ public class PartitioningScheme
     private final List<VariableReferenceExpression> outputLayout;
     private final Optional<VariableReferenceExpression> hashColumn;
     private final boolean replicateNullsAndAny;
+    private final boolean scaleWriters;
```

Comment:
can we call this scaleWriterThreads or scaleTaskWriterThreads.  Otherwise it gets to be confusing how this relates to task writer scaling (scaled_writers scheduler)

Reply:
This can actually mean both. In context of a local exchange it will be thread scaling. In context of a remote exchange it will be task scaling. Both local and remote exchanges specify partitioning requirement / preferences via the PartitioningScheme object.

Reply:
where does this get used for specifying scale writers for remote exchanges? I thought that's determined by the SCALED_WRITER_DISTRIBUTION partitioning handle.  Actually, not sure how to make this work, since the partititioningHandle for table write is a ConnectorPartitioningHandle, but it would be a bit nice for AddExchanges and AddLocalExchanges work the same way for scaling (e.g. based on the PartitioningHandle being SCALED)

Reply:
> where does this get used for specifying scale writers for remote exchanges?

We don't have an implementation for task scalling that takes partitioning information into account. For example when writing to a table with X partitions each TableWriter task will write to all X partitions. With this change at least it does not necessarily mean that each thread will be writing to all X partitions. The thread scaling algorithm will start with assigning each partition to a thread and rebalance if necessary. I think supporting remote scaling for partitioned tables would be a good next step.

For now SCALED_WRITER_DISTRIBUTION is used. But this handle does not carry any information about how data is partitioned in the underlying table. That's why we introduce the `scaleWriters` flag. With this flag the `PartitioningScheme` can contain both: the information about the table partitioning as well as a hint that the writers can be "scaled".

Reply:
thanks, that makes sense